Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs

```diff
@@ -526,99 +538,68 @@ private void EnsureMixedRealityRequirements()
 
 #region MonoBehaviour Implementation
 
-        private static MixedRealityToolkit instance;
+        private static MixedRealityToolkit activeInstance;
         private static bool newInstanceBeingInitialized = false;
 
+        /// <summary>
+        /// Lock property for the Mixed Reality Toolkit to prevent reinitialization
+        /// </summary>
+        private static readonly object initializedLock = new object();
+
         /// <summary>
         /// Returns the Singleton instance of the classes type.
         /// If no instance is found, then we search for an instance in the scene.
-        /// If more than one instance is found, we log an error and no instance is returned.
+        /// If more than one instance is found, only the first instance is activated. All others are deactivated.
         /// </summary>
         public static MixedRealityToolkit Instance
         {
             get
             {
-                if (IsInitialized)
+                if (activeInstance != null)
```

Comment:
A null check on a monobehavior is actually non-trivial due to the overridden checks.
This gets called enough that we shouldn't null check it.
https://docs.unity3d.com/Manual/BestPracticeUnderstandingPerformanceInUnity7.html

Since we no longer create any instances in the getter, I think we should slim this down and simply return whatever the active registered instance is, rather than adding any logical overhead.